.github/workflows/go.yml

```diff
@@ -11,36 +11,37 @@ on:
     - release-*
     - feature/*
 
-jobs:
-  check-changes:
-    name: Check whether tests need to be run based on diff
-    runs-on: [ubuntu-latest]
-    steps:
-    - uses: actions/checkout@v3
-      with:
-        fetch-depth: 0
-    - uses: antrea-io/has-changes@v2
-      id: check_diff
-      with:
-        paths-ignore: docs/* ci/jenkins/* *.md hack/.notableofcontents
-    outputs:
-      has_changes: ${{ steps.check_diff.outputs.has_changes }}
+env:
+  go-cache-name: go
 
+jobs:
   # test-unit-ubuntu and test-unit-windows are intentionally not merged into one job with os matrix, otherwise the job
   # wouldn't be expanded if it's skipped and the report of the required check would be missing.
   # See https://github.com/antrea-io/antrea/issues/3563.
   test-unit-ubuntu:
-    needs: check-changes
-    if: ${{ needs.check-changes.outputs.has_changes == 'yes' }}
     name: Unit test (ubuntu-latest)
     runs-on: [ubuntu-latest]
     steps:
     - name: Check-out code
       uses: actions/checkout@v3
     - name: Set up Go using version from go.mod
-      uses: actions/setup-go@v4
+      uses: actions/setup-go@v3
       with:
         go-version-file: 'go.mod'
+    - uses: actions/cache@v3
+      with:
+        # In order:
+        # * Module download cache
+        # * Build cache (Linux)
+        # * Build cache (Mac)
+        path: |
+          ~/go/pkg/mod
+          ~/.cache/go-build
+          ~/Library/Caches/go-build
+          %LocalAppData%\go-build
+        key: ${{ runner.os }}-${{ env.go-cache-name }}-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ env.go-cache-name }}-
     - name: Run unit tests
```

Comment:
**Suggestion:** Consider using a matrix strategy for the cache action to avoid duplicating code for different operating systems.
```suggestion
    jobs:
      test-unit:
        strategy:
          matrix:
            os: [ubuntu-latest, windows-2019]
        steps:
        - uses: actions/cache@v3
          with:
            path: |
              ~/go/pkg/mod
              ~/.cache/go-build
              ~/Library/Caches/go-build
              %LocalAppData%\go-build
            key: ${{ runner.os }}-${{ env.go-cache-name }}-${{ hashFiles('**/go.sum') }}
            restore-keys: |
              ${{ runner.os }}-${{ env.go-cache-name }}-
```